app/src/main/java/com/eis/phoneringer/structure/AppManager.java

```diff
@@ -0,0 +1,96 @@
+package com.eis.phoneringer.structure;
+
+import android.content.Context;
+import android.media.Ringtone;
+import android.os.Build;
+import android.os.Handler;
+import android.os.Looper;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.eis.smslibrary.SMSHandler;
+import com.eis.smslibrary.exceptions.InvalidSMSMessageException;
+import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+
+/**
+ * This is a singleton class used to manage a received RingCommand or to send one
+ *
+ * @author Alberto Ursino, Luca Crema, Alessandra Tonin, Marco Mariotto
+ */
+public class AppManager {
+
+    private static final String WRONG_PASSWORD_TOAST = "Wrong password";
+    private static final int TIMEOUT_TIME = 30 * 1000; //30 seconds
+
+    /**
+     * Instance of the class that is instantiated in getInstance method
+     */
+    private static AppManager instance = null;
+
+    /**
+     * Private constructor
+     */
+    private AppManager() {
+    }
+
+    /**
+     * @return the AppManager instance
+     */
+    public static AppManager getInstance() {
+        if (instance == null)
+            instance = new AppManager();
+        return instance;
+    }
+
+    /**
+     * If the password of the message received is valid then play ringtone for fixed amount of time
+     *
+     * @param context     of the application
+     * @param ringCommand received
+     * @param ringtone    to be played
+     */
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    public void onRingCommandReceived(Context context, @NonNull RingCommand ringCommand, final Ringtone ringtone) {
+        final RingtoneHandler ringtoneHandler = RingtoneHandler.getInstance();
+
+        if (checkPassword(context, ringCommand)) {
+            ringtoneHandler.playRingtone(ringtone);
+            //Timer: the ringtone is playing for TIMEOUT_TIME seconds.
+            new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
+                @Override
+                public void run() {
+                    ringtoneHandler.stopRingtone(ringtone);
+                }
+            }, TIMEOUT_TIME);
+        } else {
+            Toast.makeText(context, WRONG_PASSWORD_TOAST, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    /**
+     * Method used to send a RingCommand via SMS using the library class {@link SMSHandler}
+     *
+     * @param context     of the application
+     * @param ringCommand to send
+     */
+    public void sendCommand(Context context, @NonNull RingCommand ringCommand) throws InvalidSMSMessageException, InvalidTelephoneNumberException {
```

Comment:
Nah man, the two exceptions are still being thrown because this method calls `RingCommandHandler.parseCommand()` which throws them.
Also, you removed the specifications but not the "throws" after the method signature. Bring the specifications back and give them a description of when they are thrown

Reply:
I know, corrected :D

Reply:
Ok I removed the "throws" but wait, why should I add the exceptions back and specify that they are launched by the method of another class?

Reply:
Yes, you should specify that those exceptions are thrown because you make a call to `RingCommandHandler.parseCommand()` and that may throw exception

Reply:
Well ok, I put the exceptions back.
But the class RingCommandHandler? Do we leave it like this?

Reply:
For now yes, you can't edit it from this branch